diff -ruN googletest-src/googletest/cmake/internal_utils.cmake googletest-src-aurora/googletest/cmake/internal_utils.cmake
--- googletest-src/googletest/cmake/internal_utils.cmake	2019-07-16 16:34:21.513045774 +0900
+++ googletest-src-aurora/googletest/cmake/internal_utils.cmake	2019-07-16 16:35:13.530054326 +0900
@@ -99,7 +99,7 @@
     set(cxx_no_exception_flags "-EHs-c- -D_HAS_EXCEPTIONS=0")
     set(cxx_no_rtti_flags "-GR-")
   elseif (CMAKE_COMPILER_IS_GNUCXX)
-    set(cxx_base_flags "-Wall -Wshadow -Werror")
+    set(cxx_base_flags "-Wall -Werror")
     if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0.0)
       set(cxx_base_flags "${cxx_base_flags} -Wno-error=dangling-else")
     endif()
@@ -110,7 +110,7 @@
     # explicitly.
     set(cxx_no_rtti_flags "-fno-rtti -DGTEST_HAS_RTTI=0")
     set(cxx_strict_flags
-      "-Wextra -Wno-unused-parameter -Wno-missing-field-initializers")
+      "-Wno-unused-parameter -Wno-missing-field-initializers")
   elseif (CMAKE_CXX_COMPILER_ID STREQUAL "SunPro")
     set(cxx_exception_flags "-features=except")
     # Sun Pro doesn't provide macros to indicate whether exceptions and
diff -ruN googletest-src/googletest/include/gtest/internal/gtest-port.h googletest-src-aurora/googletest/include/gtest/internal/gtest-port.h
--- googletest-src/googletest/include/gtest/internal/gtest-port.h	2019-07-16 16:34:21.519045775 +0900
+++ googletest-src-aurora/googletest/include/gtest/internal/gtest-port.h	2019-07-16 16:36:26.749066361 +0900
@@ -766,7 +766,12 @@
 #   undef _TR1_FUNCTIONAL  // Allows the user to #include
                         // <tr1/functional> if they choose to.
 #  else
+#ifdef __NEC__
+#   include <tuple>  // NOLINT
+#else
 #   include <tr1/tuple>  // NOLINT
+#endif
+
 #  endif  // !GTEST_HAS_RTTI && GTEST_GCC_VER_ < 40302
 
 // VS 2010 now has tr1 support.
@@ -774,7 +779,11 @@
 #  include <tuple>  // IWYU pragma: export  // NOLINT
 
 # else  // GTEST_USE_OWN_TR1_TUPLE
+#ifdef __NEC__
+#  include <tuple>  // IWYU pragma: export  // NOLINT
+#else
 #  include <tr1/tuple>  // IWYU pragma: export  // NOLINT
+#endif
 # endif  // GTEST_USE_OWN_TR1_TUPLE
 
 #endif  // GTEST_HAS_TR1_TUPLE
