diff --git a/src/C/CMakeLists.txt b/src/C/CMakeLists.txt
index da0e022..3e7c72c 100644
--- a/src/C/CMakeLists.txt
+++ b/src/C/CMakeLists.txt
@@ -13,19 +13,19 @@ include_directories("../")
 
 add_library(vednn_c_code OBJECT
 	vednnConvolutionForward.cpp
-#	vednnConvolutionForwardAddBias.c
+	vednnConvolutionForwardAddBias.c
 	vednnConvolutionBackwardData.cpp
 	vednnConvolutionBackwardFilter.cpp
 	vednnLinearForward.cpp
 	vednnLinearBackwardData.cpp
 	vednnLinearBackwardWeight.cpp
-#	vednnActivationForward.c
-#	vednnActivationBackward.c
-#	vednnMaxPoolingForward.c
-#	vednnMaxPoolingForward_default.c
-#	vednnMaxPoolingBackward.c
-#	vednnMaxPoolingBackward_default.c
-#	vednnSoftmaxForward.c
+	vednnActivationForward.c
+	vednnActivationBackward.c
+	vednnMaxPoolingForward.c
+	vednnMaxPoolingForward_default.c
+	vednnMaxPoolingBackward.c
+	vednnMaxPoolingBackward_default.c
+	vednnSoftmaxForward.c
 	vednnInit.c
 )
 
@@ -36,4 +36,4 @@ endif()
 
 if(BUILD_SHARED_LIB)
 	target_compile_options(vednn_c_code PUBLIC -fpic)
-endif()
\ No newline at end of file
+endif()
diff --git a/src/C/vednnSoftmaxForward.c b/src/C/vednnSoftmaxForward.c
index 905ae19..eaca050 100644
--- a/src/C/vednnSoftmaxForward.c
+++ b/src/C/vednnSoftmaxForward.c
@@ -90,7 +90,7 @@ vednnError_t vednnSoftmaxForward(
 
 }
 
-static vednnError_t vednnSoftmaxForward_Fast (
+ vednnError_t vednnSoftmaxForward_Fast (
     const void 			*pDataIn,
     void 			*pDataOut,
     const uint64_t		nBatch,
@@ -119,7 +119,7 @@ static vednnError_t vednnSoftmaxForward_Fast (
   return VEDNN_SUCCESS ;
 }
 
-static vednnError_t vednnSoftmaxForward_Accurate (
+ vednnError_t vednnSoftmaxForward_Accurate (
     const void 			*pDataIn,
     void 			*pDataOut,
     const uint64_t		nBatch,
@@ -154,7 +154,7 @@ static vednnError_t vednnSoftmaxForward_Accurate (
 }
 
 
-static vednnError_t vednnSoftmaxForward_Log (
+ vednnError_t vednnSoftmaxForward_Log (
     const void 			*pDataIn,
     void 			*pDataOut,
     const uint64_t		nBatch,
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 97a7213..b629724 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -21,15 +21,15 @@ endif()
 add_library(${LIBNAME} ${LIB_MODE}
         $<TARGET_OBJECTS:vednn_c_code>
         $<TARGET_OBJECTS:vednn_intrinsic_convolution_forward>
-        #$<TARGET_OBJECTS:vednn_intrinsic_convolution_forward_addbias>
+        $<TARGET_OBJECTS:vednn_intrinsic_convolution_forward_addbias>
         $<TARGET_OBJECTS:vednn_intrinsic_convolution_backward_data>
         $<TARGET_OBJECTS:vednn_intrinsic_convolution_backward_filter>
         $<TARGET_OBJECTS:vednn_intrinsic_linear_forward>
         $<TARGET_OBJECTS:vednn_intrinsic_linear_backward_data>
         $<TARGET_OBJECTS:vednn_intrinsic_linear_backward_weight>        
-        #$<TARGET_OBJECTS:vednn_intrinsic_maxpooling_backward>
-        #$<TARGET_OBJECTS:vednn_intrinsic_maxpooling_forward>
-        #$<TARGET_OBJECTS:vednn_intrinsic_activation>
+        $<TARGET_OBJECTS:vednn_intrinsic_maxpooling_backward>
+        $<TARGET_OBJECTS:vednn_intrinsic_maxpooling_forward>
+        $<TARGET_OBJECTS:vednn_intrinsic_activation>
 )
 
 if(USE_OPENMP)
diff --git a/src/intrinsic/CMakeLists.txt b/src/intrinsic/CMakeLists.txt
index bdddd44..8047e49 100644
--- a/src/intrinsic/CMakeLists.txt
+++ b/src/intrinsic/CMakeLists.txt
@@ -16,13 +16,13 @@ endif()
 include_directories("../")
 
 add_subdirectory(Convolution/Forward)
-#add_subdirectory(Convolution/ForwardAddBias)
+add_subdirectory(Convolution/ForwardAddBias)
 add_subdirectory(Convolution/BackwardData)
 add_subdirectory(Convolution/BackwardFilter)
 add_subdirectory(Linear/Forward)
 add_subdirectory(Linear/BackwardData)
 add_subdirectory(Linear/BackwardWeight)
-#add_subdirectory(MaxPooling/Backward)
-#add_subdirectory(MaxPooling/Forward)
-#add_subdirectory(Activation)
+add_subdirectory(MaxPooling/Backward)
+add_subdirectory(MaxPooling/Forward)
+add_subdirectory(Activation)
 
diff --git a/test/Makefile b/test/Makefile
index f3c51f0..a7895de 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -44,7 +44,7 @@ CFLAGS	= $(COPTS)
 LDLIBS	= $(COPTS) -lm
 AR	= $(AURORA_BIN_DIR)/nar
 
-BLAS_DIR	= /opt/nec/ve/nlc/1.0.0
+BLAS_DIR	= /opt/nec/ve/nlc/2.3.0
 BLAS_INC_DIR	= $(BLAS_DIR)/include
 BLAS_LIB_DIR	= $(BLAS_DIR)/lib
 CFLAGS		+= -I$(BLAS_INC_DIR) 
@@ -52,8 +52,8 @@ CFLAGS		+= -I$(BLAS_INC_DIR)
 CFLAGS	+= -I${VEDNN_DIR}/include
 
 ifeq ($(OPENMP),YES)
-LDLIBS	+= -L${VEDNN_DIR}/lib -lvednn_openmp
-LDLIBS  += -L$(BLAS_LIB_DIR) -lblas_openmp
+LDLIBS	+= ${VEDNN_DIR}/lib/libvednn_openmp.a
+LDLIBS  += -L$(BLAS_LIB_DIR) -lblas_openmp  -Wl,-rpath=${BLAS_LIB_DIR}
 LDLIBS  += -fopenmp
 else
 LDLIBS	+= -L${VEDNN_DIR}/lib -lvednn_sequential
